driverapi/CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+set(
+    DRIVERAPI_SOURCES
+    ${CMAKE_SOURCE_DIR}/driverapi/src/librecuda.cpp
+    ${CMAKE_SOURCE_DIR}/driverapi/src/librecuda_status.cpp
+    ${CMAKE_SOURCE_DIR}/driverapi/src/cmdqueue.cpp
+)
+
+add_library(driverapi SHARED ${DRIVERAPI_SOURCES})
+target_include_directories(
+    driverapi PUBLIC 
+    ${CMAKE_SOURCE_DIR}/driverapi/include 
+    ${CMAKE_SOURCE_DIR}/third_party/ELFIO 
+    ${CMAKE_SOURCE_DIR}/driverapi/internal)
```

Comment:
this should be target_include_directories(driverapi PRIVATE internal/)
Also use CMAKE_CURRENT_LIST_DIR to refer to current module? That way you are not dependent on traversing from root dir.
This is resistent to moving the driverapi/ folder to a subfolder, as long as it is added to CMake, the logic will not break